hebiros_gazebo_plugin/plugin_hebiros/hebiros_gazebo_group.h

```diff
@@ -7,41 +7,50 @@
 #include "hebiros/SettingsMsg.h"
 #include "hebiros/SetCommandLifetimeSrv.h"
 #include "hebiros/SetFeedbackFrequencySrv.h"
-#include "hebiros_gazebo_joint.h"
+#include "joint.h"
 
-using namespace hebiros;
+namespace hebi {
+namespace sim {
+namespace plugin {
 
 class HebirosGazeboGroup : public std::enable_shared_from_this<HebirosGazeboGroup> {
 
 public:
+  HebirosGazeboGroup(std::string name,
+    const std::vector<hebi::sim::Joint*>& joints_,
+    std::shared_ptr<ros::NodeHandle> n);
 
-  // TODO: Make these private.
+  void UpdateFeedback(const ros::Duration& iteration_time);
+
+  ros::Time GetPrevTime() { return prev_time; }
+  void SetPrevTime(ros::Time t) { prev_time = t; }
+
+private:
   std::string name;
-  std::map<std::string, std::shared_ptr<HebirosGazeboJoint>> joints;
-  FeedbackMsg feedback;
-  CommandMsg command_target;
+  hebiros::FeedbackMsg feedback;
   bool check_acknowledgement = false;
   bool acknowledgement = false;
-  bool command_received = false;
-  bool group_added = false;
   int command_lifetime = 100;
   int feedback_frequency = 100;
 
   ros::Time start_time;
-  ros::Time prev_time;
   ros::Time prev_feedback_time;
+  ros::Time prev_time;
 
   ros::Subscriber command_sub;
   ros::Publisher feedback_pub;
   ros::ServiceServer acknowledge_srv;
   ros::ServiceServer command_lifetime_srv;
   ros::ServiceServer feedback_frequency_srv;
 
-  HebirosGazeboGroup(std::string name, std::shared_ptr<ros::NodeHandle> n);
-
-  void SubCommand(const boost::shared_ptr<CommandMsg const> data);
+  void SubCommand(const boost::shared_ptr<hebiros::CommandMsg const> data);
   bool SrvAcknowledge(std_srvs::Empty::Request &req, std_srvs::Empty::Response &res);
-  bool SrvSetCommandLifetime(SetCommandLifetimeSrv::Request &req, SetCommandLifetimeSrv::Response &res);
-  bool SrvSetFeedbackFrequency(SetFeedbackFrequencySrv::Request &req, SetFeedbackFrequencySrv::Response &res);
+  bool SrvSetCommandLifetime(hebiros::SetCommandLifetimeSrv::Request &req, hebiros::SetCommandLifetimeSrv::Response &res);
+  bool SrvSetFeedbackFrequency(hebiros::SetFeedbackFrequencySrv::Request &req, hebiros::SetFeedbackFrequencySrv::Response &res);
 
+  std::vector<hebi::sim::Joint*> joints;
```

Comment:
this may be related to my comment above regarding `unique_ptr` and "weak pointers"